extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevservicesElasticsearchBuildItem.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.elasticsearch.restclient.common.deployment;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class DevservicesElasticsearchBuildItem extends SimpleBuildItem {
```

Comment:
You'll probably need to make this a `MultiBuildItem`, otherwise Quarkus will blow up as soon as you use both the ES Low Level REST client and Hibernate Search in the same application (which is a reasonable use case).

As to how to deal with it... I'm not sure. Maybe check that at least one requests devservices, and that the versions (if any) are exactly the same, and if not throw an exception?

On the plus side, I suppose that solves the problem of multi-backend: you could deal with conflicts in the exact same way.

Reply:
The hard part of MultiBuildItem is that if it's shared by two extensions how can I know that the default Elasticsearch for the rest-client is the same as the default Elasticsearch of Hibernate search ?

We'll move to MultiBuildItem some day (maybe soon), but I prefere to use a single build item for the first implementation so when using both extension it cannot be used. By the way it make me think that the disable of devservices should be done on the rest-client/hibernate-search level not on the common lib so one can disable one of the two when using both extension as a workaround.

Reply:
> We'll move to MultiBuildItem some day (maybe soon), but I prefere to use a single build item for the first implementation so when using both extension it cannot be used.

With this solution, everything may explode as soon as both extensions (Hibernate Search and the low-level client) are used in the same application, because they will create multiple instances of a build item that is supposed to be unique. And I suspect this will lead to exceptions during the build, even if dev services are disabled.

If you just want to not support devservices for the case where two extensions contribute a `DevservicesElasticsearchBuildItem`, then that's fine. You can still make `DevservicesElasticsearchBuildItem` multi, and do runtime checks: if you get multiple such build items in `DevServicesElasticsearchProcessor.java`, you can either throw an exception asking users to disable devservices, or log a warning and disable devservices implicitly.

But please, don't introduce a new build item that will make existing application fail even if they don't use devservices :)

> By the way it make me think that the disable of devservices should be done on the rest-client/hibernate-search level not on the common lib so one can disable one of the two when using both extension as a workaround.

Yes, that's another option. If users have a switch that allows them to prevent an extension to create a `DevservicesElasticsearchBuildItem`, then everything should work. Though I suspect the error message when users enable devservices in two extensions will be ugly (basically Quarkus complaining about finding two instances of `DevservicesElasticsearchBuildItem` even though it does not extend `MultiBuildItem`).

Reply:
> If you just want to not support devservices for the case where two extensions contribute a DevservicesElasticsearchBuildItem, then that's fine. You can still make DevservicesElasticsearchBuildItem multi, and do runtime checks: if you get multiple such build items in DevServicesElasticsearchProcessor.java, you can either throw an exception asking users to disable devservices, or log a warning and disable devservices implicitly.

Yes, a simple implementation can be to migrate it to a Multi build item and don't start devservices (or start the first one) and log an error.

> Yes, that's another option. If users have a switch that allows them to prevent an extension to create a DevservicesElasticsearchBuildItem, then everything should work. 

I think this is orthogonal to the issue, both must be done.

Reply:
I switch to a MultiBuildItem and add a check that throws a BuildException for now.
I think if both extensions are present with not default, we should detect this case and treat it as valid then emit both property. I'll let this for a followup PR as I want this to be in the next release so I prefere we finish this one first.